### PR TITLE
Fix hook for Doctrine PDO class

### PR DESCRIPTION
**Pull request by matikij**
When using doctrine the return type of PDO->prepare is Doctrine\DBAL\Driver\PDO\Statement and the hook didn't work. Changed the if to a match statement and added a bunch of debug statements. Fixed a minor typo.

### CODE REVIEW

**Comment by jmjoy** [comment]
Please format the code with: `cargo +nightly fmt --all`

File: src/plugin/plugin_pdo.rs [hunk 1]
```diff
@@ -198,8 +198,17 @@ fn after_hook(
             );
         }
     } else if let Some(obj) = return_value.as_mut_z_obj() {
-        if obj.get_class().get_name() == &"PDOStatement" {
-            return after_hook_when_pdo_statement(get_this_mut(execute_data)?, obj);
+        let cls=obj.get_class().get_name().to_str()?;
+        match cls {
+            "PDOStatement" => {
```
Comment (jmjoy) [inline src/plugin/plugin_pdo.rs:203]: Use `"PDOStatement" | r"Doctrine\DBAL\Driver\PDO\Statement"` here so you can merge the condition?

File: src/plugin/plugin_pdo.rs [hunk 2]
```diff
@@ -198,8 +198,17 @@ fn after_hook(
...
+                return after_hook_when_pdo_statement(get_this_mut(execute_data)?, obj);
+            }
+            r"Doctrine\DBAL\Driver\PDO\Statement" => {
+                return after_hook_when_pdo_statement(get_this_mut(execute_data)?, obj);
+            }
+            _ => {
+                debug!(cls, "not a valid class");
+            }
```
Comment (jmjoy) [inline src/plugin/plugin_pdo.rs:211]: Is `Doctrine\DBAL\Driver\PDO\Statement` the subclass of `PDOStatement`?
Maybe it's better to [is_instance_of](https://docs.rs/phper/latest/phper/classes/struct.ClassEntry.html#method.is_instance_of) to detect.

Comment (matikij) [reply src/plugin/plugin_pdo.rs]: Indeed this is a much better approach and it works for my case. Addressed in next commit, run cargo fmt and went back from match to if/else

Comment (jmjoy) [reply src/plugin/plugin_pdo.rs]: Good job!

**Review by jmjoy — APPROVED** [review]
LGTM

**Pull request merged**